blackbox/blackbox/httpjob.go

```diff
@@ -0,0 +1,83 @@
+package blackbox
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type Job struct {
+	Name       string        `yaml:"job_name"`
+	Targets    []url.URL     `yaml:"targets"`
+	Timeout    time.Duration `yaml:"timeout"`
+	PeriodTime time.Duration `yaml:"period_time"`
+}
+
+func (j Job) IsValid() error {
+	if j.Timeout == 0 {
+		return fmt.Errorf("job is not valid: timeout is 0")
+	}
+
+	if j.PeriodTime == 0 {
+		return fmt.Errorf("job is not valid: period time is 0")
+	}
+	return nil
+}
+
+// Do scrapes all target from job's targets and creates Result from them. This
+// Result is passed to scraper reporter using r channel.
+func (j Job) Do(r chan<- Result) {
+	var mu sync.Mutex
+	var responses []http.Response
+	ticker := time.Tick(j.PeriodTime)
+
+	for range ticker {
+		responses = make([]http.Response, 0, len(j.Targets))
+		ctx, cancel := context.WithTimeout(context.Background(), j.Timeout)
+		g, ctx := errgroup.WithContext(ctx)
+		for _, t := range j.Targets {
+			scrapedURL := t
+			g.Go(func() error {
+				resp, err := httpScrape(ctx, scrapedURL, j.Timeout)
+				if err != nil {
+					return err
+				}
+				mu.Lock()
```

Comment:
this is quite unusual to have mutex together with chan for collecting results
imho - worth changing Result a bit to be able to collect the parts of job and combine it to one job if needed